src/rf/apps/workers/process.py

```diff
@@ -0,0 +1,221 @@
+                    self.q.delete_message(message)
+
+    def validate_image(self, data):
+        """
+        Use Gdal to verify an image is properly formatted and can be further
+        processed.
+        data -- attribute data from SQS.
+        """
+        # Pass image to Gdal to verify.
+        # If it succeeds continue else return False.
+
+        key = data['s3']['object']['key']
+        dot = key.find('.') if key.find('.') >= 0 else len(key)
+        first_dash = key.find('-')
+        payload_uuid = key[first_dash:dot]
+
+        url = self.make_s3_url(data['s3']['bucket']['name'], key)
+
+        # TODO - Right now we don't save data to the DB with upload. So hard
+        # code something for testing.
+        try:
+            image = LayerImage.objects.get(s3_uuid=payload_uuid)
+            image.status = STATUS_VALIDATED
+            image.save()
+        except:
+            payload_uuid = '1aa064aa-1086-4ff1-a90b-09d3420e0343'
+
+        data = {'url': url, 's3_uuid': payload_uuid}
+        payload = self.make_payload(JOB_REPROJECT, data)
+        self.post_to_queue(payload)
+        return True
+
+    def reproject(self, data):
+        """
+        Reproject incoming image to Web Mercator.
+        data -- attribute data from SQS.
+        """
+        # Pass image to Gdal to reproject
+        # If it succeeds continue else return False.
+
+        s3_uuid = data['s3_uuid']
+        image = LayerImage.objects.get(s3_uuid=s3_uuid)
+        image.status = STATUS_REPROJECTED
+        image.save()
+
+        data = {'layer_id': image.layer_id}
+        payload = self.make_payload(JOB_HANDOFF, data)
+        self.post_to_queue(payload)
+        return True
+
+    def emr_hand_off(self, data):
+        """
+        Passes layer imgages to EMR to begin creating custom rasters.
+        data -- attribute data from SQS.
+        """
+        # If all the images are ready send data to EMR for processing.
+        # return False if it fails to start.
+        # EMR posts directly to queue upon competion.
+
+        layer_id = data['layer_id']
+        layer_images = LayerImage.objects.filter(layer_id=layer_id)
+        reprojected_images = LayerImage.objects.filter(layer_id=layer_id,
+                                                       status=STATUS_REPROJECTED)  # NOQA
+        ready_to_process = len(layer_images) == len(reprojected_images)
+
+        if ready_to_process:
+            layer_meta = LayerMeta.objects.get(layer_id=layer_id)
+            layer_meta.state = STATUS_PROCESSING
+            layer_meta.save()
+
+            # POST TO EMR HERE.
+
+            # Add a message to the queue that we can use to watch for
+            # EMR timeout.
+            data = {
+                'timeout': time.time() + TIMEOUT_SECONDS,
+                'layer_id': layer_id
+            }
+
+            payload = self.make_payload(JOB_TIMEOUT, data)
+            self.post_to_queue(payload)
+
+        return True
```

Comment:
Now that I look closer, it looks like `reproject` job will queue an `emr_hand_off` job for each uploaded image. I don't think this will work in all cases.

For example, if you create a layer with 2 images but 1 fails to upload to S3, then only 1 set of verify/reproject/emr_hand_off jobs will queue. The result is that the layer will be in a limbo state, where it will never be marked as either completed or failed.

I think we'll need something similar to the `check_timeout` job, but for verifying that the layer is ready to be handed off. It could be created when a layer is added to the database, and continuously check if all the images have been processed before queuing the hand off job.